### PR TITLE
[Tests-Only] Add webui tests for various file operations by public in public share

### PR DESCRIPTION
**Pull request by HariBhandari07**
## Description

This PR adds webui tests for
1. folder creation in public share
2. batch deletion in public share
3. file/folder selection and clear selection in files page and in public share
4. assert that `roles dropdown of public share` changes in accordance with the user's selection

## Related Issue

- Fixes https://github.com/owncloud/web/issues/4426

## How Has This Been Tested?

- CI
- Manual
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature [hunk 1]
```diff
@@ -793,3 +793,67 @@ Feature: Share by public link
...
+    And the public marks these files for batch action using the webUI
+      | name                |
+      | lorem.txt           |
+      | simple-empty-folder |
+      | data.zip            |
+    And the public batch deletes the marked files using the webUI
+    Then file "data.zip" should not be listed on the webUI
+    And folder "simple-folder" should not be listed on the webUI
+    And file "lorem.txt" should not be listed on the webUI
+    And as "user1" folder "/simple-folder/simple-folder" should not exist
```
Comment (jasson99) [inline tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature:824]:
```suggestion
    And as "user1" folder "/simple-folder/simple-empty-folder" should not exist
```

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js [hunk 2]
```diff
@@ -231,11 +231,34 @@ module.exports = {
...
+          assert.strictEqual(
+            result.value,
+            status,
+            status === result
+              ? path + ' is expected to be selected but is not selected'
+              : path + ' is expected not to be selected but is selected'
+          )
+        })
+      })
+    },
```
Comment (dpakach) [inline tests/acceptance/pageObjects/FilesPageElement/filesList.js:261]: It's better idea to move assertions away from page objects as we discussed a while back

**Review by dpakach — CHANGES_REQUESTED** [review]

File: tests/acceptance/pageObjects/FilesPageElement/filesList.js [hunk 3]
```diff
@@ -231,11 +231,34 @@ module.exports = {
...
+        if (!result.value.ELEMENT) {
+          return
+        }
+        this.api.elementIdSelected(result.value.ELEMENT, result => {
+          assert.strictEqual(
+            result.value,
+            status,
+            status === result
+              ? path + ' is expected to be selected but is not selected'
+              : path + ' is expected not to be selected but is selected'
```
Comment (jasson99) [inline tests/acceptance/pageObjects/FilesPageElement/filesList.js:257]: I think this always returns the second message. Is status expected to be equal with the result ever?

**Review by jasson99 — CHANGES_REQUESTED** [review]

File: tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js [hunk 4]
```diff
@@ -36,12 +36,19 @@ module.exports = {
...
+      )
       return this.waitForElementPresent('@selectRoleButton')
         .click('@selectRoleButton')
         .waitForElementVisible('@rolesDropdown')
         .waitForElementVisible(`@role${role}`)
         .click(`@role${role}`)
         .waitForElementNotVisible('@rolesDropdown')
+        .useXpath()
+        .waitForElementVisible(selectedRoleDropdown)
+        .useCss()
```
Comment (HariBhandari07) [inline tests/acceptance/pageObjects/FilesPageElement/publicLinksDialog.js:51]: This is required to check the `role dropdown of public share` changes when a user selects various roles. This asserts that PR https://github.com/owncloud/web/pull/4504   is working as intended.

**Review by dpakach — COMMENTED** [review]
lgtm 👍🏽

**Review by jasson99 — COMMENTED** [review]
:+1:

**Review by phil-davis — APPROVED** [review]

**Pull request merged**